### ISSUE TITLE
access_elem failed

### ISSUE THREAD

**Issue by emilk** [raskr/rust-autograd#33]
Hi!

I am just starting to explore `autograd`, and I ran into an issue when using a 2D input:

``` rust
    autograd::with(|g| {
        use autograd::ndarray;
        let (w, h) = (2, 2);
        let x = g.placeholder(&[w as isize, h as isize]);
        let loss = x.access_elem(0).access_elem(0) * x.access_elem(0).access_elem(0)
            + x.access_elem(0).access_elem(1) * x.access_elem(0).access_elem(1);

        let grad = g.grad(&[loss], &[x])[0]; // dloss/dx

        let x_values = ndarray::Array2::from_shape_vec((w, h), vec![0.0; w * h]).unwrap();
        let _grad = grad
            .eval(&[x.given(x_values.view())])
            .expect("Failed to eval gradient");
    });
```

This gives me `Failed to eval gradient: OpError(OutOfBounds("access_elem failed."))`.

Whichever shape I pick (`1,2`, `2,1`, `2,2` etc) I get the same error.

**Comment by emilk** [comment]
Oh - I just realized `access_elem` is using flat indexing. That explains it!
